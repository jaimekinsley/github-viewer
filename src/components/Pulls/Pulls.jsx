import React from 'react';
import PropTypes from 'prop-types';

const Pulls = ({ pulls }) => {
  const pullRequests = pulls.map(pull => (
    <li key={`${pull.html_url}+${pull.numer}`}>
      <a href={pull.html_url}> {pull.number} </a>
    </li>
  ));

  return (
    <ul>
      {pullRequests}
    </ul>
  );
};

Pulls.propTypes = {
  pulls: PropTypes.arrayOf(PropTypes.object)
};

export default Pulls;
