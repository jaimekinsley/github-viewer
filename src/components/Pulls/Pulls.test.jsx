import React from 'react';
import { shallow } from 'enzyme';
import Pulls from './Pulls';

describe('Pulls component', () => {
  it('renders Pulls', () => {
    const wrapper = shallow(<Pulls pulls={[
      { number: 1 },
      { number: 2 }
    ]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
