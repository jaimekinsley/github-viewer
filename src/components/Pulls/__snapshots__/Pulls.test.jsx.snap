// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pulls component renders Pulls 1`] = `
<ul>
  <li
    key="undefined+undefined"
  >
    <a>
       
      1
       
    </a>
  </li>
  <li
    key="undefined+undefined"
  >
    <a>
       
      2
       
    </a>
  </li>
</ul>
`;
